jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceListing.java

```diff
@@ -84,12 +94,12 @@ public static String getAsHTML(Resource resource, String base, boolean parent, S
         }
 
         // Perform sort
-        Comparator<? super Resource> sort = switch (sortColumn)
-        {
-            case "M" -> ResourceCollators.byLastModified(sortOrderAscending);
-            case "S" -> ResourceCollators.bySize(sortOrderAscending);
-            default -> ResourceCollators.byName(sortOrderAscending);
-        };
+        Comparator<? super Path> sort = switch (sortColumn)
```

Comment:
Path cannot be used here as all directory entries mighty not have paths. A subdir might itself be a resource collection. Currently that is erroneously returning the path of the first item in the collection, that may have a totally unrelated name.

Reply:
This comment is on a `Comparator`, what specific thing are you wanting to point out?

This is a `ResourceListing`, so it should only be presenting 1 level of information deep.

Reply:
> Path cannot be used here as all directory entries mighty not have paths.

A `ResourceCollection` can have non-directory entries, correct.
But every one of those entries is a `PathResource` implementation, hence it has a `Path`.

I just added another test for this scenario (a `ResourceCollection` consisting of file, then dir, then a different dir)

Reply:
Reverting Path handling in this PR, in favor of using the work from PR #5142 
Keeping this PR's XHTML changes though.